src/util.rs

```diff
@@ -218,6 +222,16 @@ pub(crate) async fn start_accepting_connections(
             }
         }
     });
+    debug_assert_ne!(resolved_addr.port(), 0);
+    *port = resolved_addr.port();
+    let resolved_host: Host = resolved_addr.ip().into();
+    if let Host::Ipv4(ip) = host {
+        debug_assert!(ip == &mut resolved_addr.ip());
+        *host = resolved_host;
+    } else if let Host::Ipv6(ip) = host {
+        debug_assert!(ip == &mut resolved_addr.ip());
+        *host = resolved_host;
```

Comment:
I think there are ipv6 addresses that map to ipv4 addresses and I wasn't sure if `local_addr` might convert the ipv6 one to ipv4 in those cases. If it did, the `Host` variant would change from ipv6 to ipv4 in those instances. I am not sure if `local_addr` actually does that transformation, or if its a bug or feature.